src/c2ffi/asdf.lisp

```diff
@@ -0,0 +1,162 @@
+                :accessor c2ffi-file/trace-c2ffi)
+   (ffi-name-transformer :initarg :ffi-name-transformer
+                         :initform 'default-ffi-name-transformer
+                         :accessor c2ffi-file/ffi-name-transformer)
+   (ffi-type-transformer :initarg :ffi-type-transformer
+                         :initform 'default-ffi-type-transformer
+                         :accessor c2ffi-file/ffi-type-transformer)
+   (prelude :initform nil
+            :initarg :prelude
+            :accessor c2ffi-file/prelude)
+   (foreign-library-name :initarg :foreign-library-name
+                         :initform nil
+                         :accessor c2ffi-file/foreign-library-name)
+   (foreign-library-spec :initarg :foreign-library-spec
+                         :initform nil
+                         :accessor c2ffi-file/foreign-library-spec)
+   (exclude-archs :initarg :exclude-archs
+                  :initform nil
+                  :accessor c2ffi-file/exclude-archs)
+   (sys-include-paths :initarg :sys-include-paths
+                      :initform nil
+                      :accessor c2ffi-file/sys-include-paths)
+   (emit-generated-name-mappings :initarg :emit-generated-name-mappings
+                                 :initform :t
+                                 :accessor c2ffi-file/emit-generated-name-mappings)
+   (include-sources :initarg :include-sources
+                    :initform :all
+                    :accessor c2ffi-file/include-sources)
+   (exclude-sources :initarg :exclude-sources
+                    :initform nil
+                    :accessor c2ffi-file/exclude-sources)
+   (include-definitions :initarg :include-definitions
+                        :initform :all
+                        :accessor c2ffi-file/include-definitions)
+   (exclude-definitions :initarg :exclude-definitions
+                        :initform nil
+                        :accessor c2ffi-file/exclude-definitions))
+  (:default-initargs
+   :type nil)
+  (:documentation
+   "The input of this ASDF component is a C header file, plus configuration for the
+binding generation process. This header file will define the initial scope of
+the generation process, which can further be filtered by other configuration
+parameters.
+
+An external program based on clang/llvm called 'c2ffi' is then invoked with
+this header file to generate a json file for various platforms. The generation
+of the underlying platform's json file must succeed to continue, but the
+generation for the other platforms is allowed to fail
+\(see ENSURE-SPEC-FILE-EXISTS).
+
+It's advisable to deliver these json files with the project, so that users
+don't need to have c2ffi installed.
+
+Once/if the json file is available for the underlying platform, then the json
+file is used to generate a lisp file with CFFI definitions (see
+PROCESS-C2FFI-SPEC-FILE). This file will then be compiled as any other lisp
+file, except that it's will be stored in the fasl cache."))
```

Comment:
OK, I'll try and tackle the manual bits. I suppose it'll help me understand the code better. I've opened https://bugs.launchpad.net/cffi/+bug/1538781 to track that. I hope it doesn't sit there forgotten forever. :(